synapse/handlers/account_validity.py

```diff
@@ -65,15 +60,16 @@ def __init__(self, hs):
 
             self._raw_from = email.utils.parseaddr(self._from_string)[1]
 
-            self._template_html, self._template_text = load_jinja2_templates(
-                self.config.email_template_dir,
+            self._template_html, self._template_text = hs.config.read_templates(
                 [
                     self.config.email_expiry_template_html,
                     self.config.email_expiry_template_text,
                 ],
-                apply_format_ts_filter=True,
-                apply_mxc_to_http_filter=True,
-                public_baseurl=self.config.public_baseurl,
+                self.config.email_template_dir,
+                filters={
+                    "format_ts": format_ts_filter,
+                    "mxc_to_http": create_mxc_to_http_filter(hs.config.public_baseurl),
+                },
```

Comment:
I don't think there's much cost to including new filters really, so we could include these for all templates. The issue is if multiple calling functions want to use the same filter with different arguments, then a `filter` argument to `read_templates` would still make sense.

But... at the moment we don't need that, so the `filters` argument could potentially be removed.

Reply:
> The issue is if multiple calling functions want to use the same filter with different arguments, then a `filter` argument to `read_templates` would still make sense.

You mean if different templates want to use a filter name that does different things? I would argue that we should not do that, we should be consistent as a project.

Reply:
I'm referring to something like the `create_mxc_to_http_filter` function, which returns a filter, but takes an argument. If we needed a similar function that requires a custom argument, perhaps depending on which endpoint was calling `read_templates`, then it would make sense for such a function to be passed to `read_templates`, perhaps using the `filters` argument similar to what we have here.

But, as I said that's not a requirement at the moment and may never be, so it may make sense to remove `filters` from `read_templates` for now until it's needed?

Reply:
> I'm referring to something like the `create_mxc_to_http_filter` function, which returns a filter, but takes an argument.

Ah -- but the argument is always the same for a running Synapse, right?

I think keeping it simple until we need that feature is the way to go.